ci-operator/step-registry/workers-rhel/upgrade/run/workers-rhel-upgrade-run-commands.sh

```diff
@@ -0,0 +1,85 @@
+        echo -e "oc get clusterversion/version -oyaml\n$(oc get clusterversion/version -oyaml)"
+        echo -e "Describing abnormal nodes...\n"
+        oc get node --no-headers | awk '$2 != "Ready" {print $1}' | while read node; do echo -e "\n#####oc describe node ${node}#####\n$(oc describe node ${node})"; done
+        echo -e "Describing abnormal operators...\n"
+        oc get co --no-headers | awk '$3 != "True" || $4 != "False" || $5 != "False" {print $1}' | while read co; do echo -e "\n#####oc describe co ${co}#####\n$(oc describe co ${co})"; done
+        echo -e "Describing abnormal mcp...\n"
+        oc get mcp --no-headers | awk '$3 != "True" || $4 != "False" || $5 != "False" {print $1}' | while read mcp; do echo -e "\n#####oc describe mcp ${mcp}#####\n$(oc describe mcp ${mcp})"; done
+    fi
+}
+
+# Generate the Junit for RHEL upgrade
+function createUpgradeJunit() {
+    echo "Generating the Junit for RHEL upgrade"
+    if (( FRC == 0 )); then
+      cat >"${ARTIFACT_DIR}/junit_rhel_upgrade.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="RHEL upgrade" tests="1" failures="0">
+  <testcase classname="RHEL upgrade" name="RHEL nodes upgrade should succeed"/>
+</testsuite>
+EOF
+    else
+      cat >"${ARTIFACT_DIR}/junit_rhel_upgrade.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="RHEL upgrade" tests="1" failures="1">
+  <testcase classname="RHEL upgrade" name="RHEL nodes upgrade should succeed"/>
+    <failure message="">RHEL nodes upgrade failed</failure>
+  </testcase>
+</testsuite>
+EOF
+    fi
+}
+
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    else
+        echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+# Install an updated version of the client
+mkdir -p /tmp/client
+curl https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/linux/oc.tar.gz | tar --directory=/tmp/client -xzf -
+PATH=/tmp/client:$PATH
+oc version --client
+
+echo "$(date -u --rfc-3339=seconds) - Validating parsed Ansible inventory"
+ansible-inventory -i "${SHARED_DIR}/ansible-hosts" --list --yaml
+echo "$(date -u --rfc-3339=seconds) - Running RHEL worker upgrade"
+ansible-playbook -i "${SHARED_DIR}/ansible-hosts" playbooks/upgrade.yml -vvv
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+echo "$(date -u --rfc-3339=seconds) - Check K8s version on the RHEL node"
+master_0=$(oc get nodes -l node-role.kubernetes.io/master -o jsonpath='{range .items[0]}{.metadata.name}{"\n"}{end}')
+rhel_0=$(oc get nodes -l node.openshift.io/os_id=rhel -o jsonpath='{range .items[0]}{.metadata.name}{"\n"}{end}')
+exp_version=$(oc get node ${master_0} --output=jsonpath='{.status.nodeInfo.kubeletVersion}' | awk -F '+' '{print $1}')
```

Comment:
we may only need to compare the major k8s version here, since sometimes the "kubernetes" rpm package shipped for RHEL worker is not exactly the same with the RHCOS one, such as:
`v1.21.1+66b664d` vs. `v1.21.0-rc.0+1622f87`
As long as the RHEL worker was upgraded to the cluster version, the major k8s version `v1.21`/ `v1.22` would be the same. 
So I guess 
`oc get node ${master_0} --output=jsonpath='{.status.nodeInfo.kubeletVersion}' | cut -d '.' -f 1,2` would work better.

Reply:
Woohoo! Thanks for the suggestion and details!! Let me update.